src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1480,24 +1480,28 @@ private static Pair<DiskAccessMode, Boolean> resolveCommitLogWriteDiskAccessMode
     {
         boolean compressOrEncrypt = getCommitLogCompression() != null || (getEncryptionContext() != null && getEncryptionContext().isEnabled());
         boolean directIOSupported = false;
-        try
+        // File.getBlockSize creates directories/files tools may not have permissions for
+        if (!toolInitialized)
```

Comment:
@jonmeredith  can you just `return` when `toolInitialized` is true? Then we do not need to have all that logic in `if`.